sotodlib/tod_ops/demodulation.py

```diff
@@ -0,0 +1,264 @@
+        if np.max(np.abs(aman.signal[i][start: stop])) < threshold:
+            if (plot):
+                plt.plot(relative_time[start: stop],
+                         aman.signal[i][start: stop])
+
+            phase_fit_signals.append(aman.signal[i][start: stop])
+    if (plot):
+        plt.grid()
+        plt.show()
+
+    return phase_fit_signals, aman.timestamps[start: stop]
+
+
+def _fit_sin(tt, yy, freq=8, plot=False):
+    '''Fit sin to the input time sequence, and return fitting parameters
+    "amplitude", "omega", "phase", "offset", "freq"'''
+    # Create sine func to fit.
+    omega = 2 * np.pi * freq
+
+    def sinfunc(t, A, phi, c):
+        return np.abs(A) * np.sin(omega * t + phi) + c
+
+    tt = np.array(tt)
+    yy = np.array(yy)
+    # guess the amplitude and offset of this sine wave
+    guess_amp = np.std(yy) * 2.**0.5
+    guess_offset = np.mean(yy)
+    guess = np.array([guess_amp, 0., guess_offset])
+
+    # fit to the data.
+    popt, pcov = curve_fit(sinfunc, tt, yy, p0=guess, maxfev=10000)
+
+    A, phi, c = popt
+
+    if (plot):
+        plt.plot(tt, yy, label='data')
+        plt.plot(tt, sinfunc(tt, *popt), label='fit, phase = %.2f' % phi)
+        plt.legend()
+        plt.show()
+
+    return {"amplitude": A, "phase": phi, "offset": c, "freq": freq}
+
+
+def fit_phase(aman, middle_relative_time, index_limit=180, threshold=.8,
+              freq=8.0, plot=False):
+    '''Fit a sinewave to the data to find the chopper phase.
+
+    args:
+
+        aman                 - axis manager to use
+        middle_relative_time - (relative) time since the beginning of the
+                               timestream that we want to fit around
+        index_limit          - +/- limits of indices around
+                               middle_relative_time that we look
+        threshold            - power threshold that we cut off when checking
+                               various detectors (if we should use their data)
+        freq                 - chopper frequency (Hz)
+        plot                 - show plots of data
+
+    returns a phase value to use with demod_single_sine
+
+    '''
+    phase_fit_signals, times = get_phase_fit_signals(
+        aman, middle_relative_time, threshold=threshold,
+        index_limit=index_limit, plot=plot)
+
+    fit_attrs = [_fit_sin(times - times[0], signal, freq=freq,
+                          plot=False) for signal in phase_fit_signals]
+    phases = np.array([attrs['phase'] for attrs in fit_attrs])
+    # convert this phase to a global phase
+    global_phases = phases - (2 * np.pi * freq) * (
+        times[0] - aman.timestamps[0])
+    # convert to usual range
+    global_phases = np.unwrap(np.sort(np.mod(global_phases, 2 * np.pi)))
+    mean, median, std = (np.mean(global_phases), np.median(global_phases),
+                         np.std(global_phases))
+
+    print('mean, median, std of fitted phases: %.2f, %.2f, %.2f' % (
```

Comment:
use logging for print statements instead of just going to sysout (you can copy what's in `sotodlib.io.load_smurf`)

Reply:
done. Is logger.debug the right choice here?